### PR TITLE
(#24458) feat: DAG_DISCOVERY_SKIP_CHECK_NAME added

### PR DESCRIPTION
**Pull request by seunggabi**
## #24458 
- [x] added `DAG_DISCOVERY_SKIP_CHECK_NAME`
- [x] test code

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by seunggabi** [comment]
@potiuk please check! thx

**Comment by eladkal** [comment]
The optimization of `DAG_DISCOVERY_SAFE_MODE` makes alot of sense. I don't know of a way to write a dag without using the words dag and airflow (due to needed imports).

If the issue is that we want also to consider these words in file names then I think It is better to extend the current setting rather than introducing new one.

from my experience dag discovery issues are one lf the hardest to figure out for Airflow users (sometimes even experianced ones) this is shared between cluster admins and DAG authors (which are not always the same person in big organizations) I even talk about this exact problem in meetups. By accepting this PR we will even make the problem bigger as it now will involve 2 settings.

Reading the discussion thread linked to the PR I didn't see any advantage for adding this setting. The `DAG_DISCOVERY_SAFE_MODE` had a big optimization advantage. What advantage are we getting here?

For the moment I'm -1 for this but lets discuss and see...

**Comment by seunggabi** [comment]
@eladkal 
Thx to your comment.
I agree with you, but I think we need to skip_check_name.

If we make module extends airflow library(e. g. DefaultDAG) , class extended this default_dag.py dosent have airflow keyword.

So, we need to # airflow comment.. 

It is bad, 
Thats why we need file or dir name checking, including dag keyword.

I suggested including skip_check_name in safe_mode.
Because another config value is not good.
(Confuse, abuse, many context in the future)

It is compatible, because it extend functions.

Please give me opinion, have a nice weekend.

File: docs/apache-airflow/concepts/dags.rst [hunk 1]
```diff
@@ -132,6 +132,8 @@ While both DAG constructors get called when the file is accessed, only ``dag_1``
 
     To consider all Python files instead, disable the ``DAG_DISCOVERY_SAFE_MODE`` configuration flag.
 
+    To consider ``dir_name`` or ``file_name`` have ``dag`` keyword files. disable the ``DAG_DISCOVERY_SKIP_CHECK_NAME`` configuration flag.
```
Comment (uranusjr) [inline docs/apache-airflow/concepts/dags.rst:135]: > To consider ``dir_name`` or ``file_name`` have ``dag`` keyword files

I’m not able to follow what this sentence means.

File: airflow/models/dagbag.py [hunk 2]
```diff
@@ -95,6 +95,7 @@ def __init__(
         read_dags_from_db: bool = False,
         store_serialized_dags: Optional[bool] = None,
         load_op_links: bool = True,
+        skip_check_name: bool = conf.getboolean('core', 'DAG_DISCOVERY_SKIP_CHECK_NAME'),
```
Comment (ashb) [inline airflow/models/dagbag.py:98]: We shouldn't be adding a new flag for this -- there is alread a setting called "safemode".

**Review by ashb — CHANGES_REQUESTED** [review]
We shouldn't be adding a new flag for this -- there is alread a setting called "safe_mode".

**Comment by ashb** [comment]
```
safe_mode=conf.getboolean('core', 'DAG_DISCOVERY_SAFE_MODE'), 
```

This is not documente d(other that the comment in the config file)

I'm closing this PR as the feature already exists -- @seunggabi Could you open a docs PR instead?

**Comment by ashb** [comment]
A second argument doesn't make any sense to me. It is already parsing files *in the `dags` folder!*.  If it's not a dag, don't put it in that folder.

**Comment by seunggabi** [comment]
@ashb 
Got it thx 👍

**Pull request closed without merging**